src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=ba0f40356d1023066d960f6de2be1a4b&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Expanded widget
+
+Similar to the `Flexible` widget, the `Expanded` widget can wrap a widget and force the widget to fill extra space. 
+
+{{site.alert.tip}}
+  **What's the difference between Flexible and Expanded?**
+    Use `Flexible` to resize widgets in a `Row` or `Column`,
+    so you can adjust child widgets' spacing while keeping the
+    relation with their parent widget.
+    `Expanded` changes the constraints of the child widgets
+    to fill any empty space.
+{{site.alert.end}} 
+
+#### Example: Filling extra space
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example demonstrates how the `Expanded` widget forces its child widget to fill extra space.
+
+  **1.** Click the **Run** button. 
+
+  **2.** Wrap the second `BlueBox` widget in an `Expanded` widget. 
+
+  For example: 
+
+  ```dart
+  Expanded(child: BlueBox(),),
+  ```
+  **3.** Select the **Format** button, and run again. 
+
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/77021d2ed15f9ece850de15e73c47526
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=77021d2ed15f9ece850de15e73c47526&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## SizedBox widget
+
+The `SizedBox` widget can be used in one of two ways.
+When `SizedBox` wraps a widget, 
+it resizes the widget using the `height` and `width` properties.
+When it doesn't wrap a widget, 
+it uses the `height` and `width` properties to create empty space.
+#### Example: Resizing a widget 
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example wraps the middle `BlueBox` widget inside of a
+  `SizedBox` widget and sets the `BlueBox`'s width to 100 logical pixels.
+
+  **1.** Click the **Run** button.
+
+  **2.** Add a `height` property equal to 100 logical pixels inside the `SizedBox` widget, and run again.   
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/6582851e85b57180ff5321f814fabb81
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=6582851e85b57180ff5321f814fabb81&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+#### Example: Creating space 
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example contains three `BlueBox` widgets and one `SizedBox` widget that separates the first and second `BlueBox` widgets. The `SizedBox` widget contains a `width` property equal to 50 logical pixels. 
+
+  **1.** Click the **Run** button.
+
+  **2.** Create more space by adding another `SizedBox` widget (25 logical pixels wide) 
+  between the second and third `BlueBox` widgets, and run again.
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/19ead147ab5c7668d7d32e1cfed90097
```

Comment:
Btw the contrast of the gray on green messages is pretty low. I should file a bug if there isn't already one.

Reply:
Interesting, I wonder why those colors were picked because the light DartPads display the messages in bright green and tan.